### ISSUE TITLE
softmax focal loss

### ISSUE THREAD

**Issue by mehran66** [qubvel/segmentation_models.pytorch#671]
The FocalLoss in this repo refer to the focal_loss_with_logits for both the binary and multiclass cases, though the multiclass should use the softmax_focal_loss_with_logits fucntion in the defined in the _functional.py.

https://github.com/qubvel/segmentation_models.pytorch/blob/c48c906bc2ee238f45aedf413e9248c37f088894/segmentation_models_pytorch/losses/focal.py
